cli/compose/convert/compose.go

```diff
@@ -101,18 +101,17 @@ func Secrets(namespace Namespace, secrets map[string]composetypes.SecretConfig)
 			continue
 		}
 
-		data, err := ioutil.ReadFile(secret.File)
+		if secret.Name != "" {
+			name = secret.Name
+		} else {
+			name = namespace.Scope(name)
+		}
```

Comment:
This if/else block can be moved into `fileObjectConfig()` to remove the duplication.